packages/runtime/container-runtime/src/opLifecycle/opGroupingManager.ts

```diff
@@ -12,19 +12,21 @@ interface IGroupedMessage {
 	contents?: unknown;
 	metadata?: Record<string, unknown>;
 	compression?: string;
+	numEmptyToCreate?: number;
```

Comment:
I think the implementation assumes that compression is optional, and ops grouping could be applied on top of uncompressed payload.
I wonder if we should assume that some sort of compression is always ON, and that process does grouping. In other words, there is no need for this layer at all, as compression process always produced just one op.

The downside of that would be - we might not have enough content or desire to run CPU-intensive compression / decompression (maybe in future we will rely on websocket protocol compression). But I think that has simple answer - we can leverage noop compression, i.e., process that groups content of all ops into one buffer but does not do actual compression.

The reasons I'm poking at it:
- less layers, less code
- while compression might be more optional, I feel grouping should become standard (always on), as it has direct impact on COGS and user costs (AFR) or throttling. It feels like both compression layer and grouping do the same thing - they both do op grouping, but with slightly different outcomes. It would be nice for "core" compression layer to do simple thing - compress single op. To some extend I'm saying - move grouping earlier in op sending pipeline - group first (result - single op), compress one op after that, then chunk. Make compression layer trivial and care about only one thing - compressing a buffer.